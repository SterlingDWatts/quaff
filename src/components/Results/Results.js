import React from "react";
import QuizContext from "../../contexts/QuizContext";
import QuizListContext from "../../contexts/QuizListContext";
import ResultsDisplay from "../ResultsDisplay/ResultsDisplay";
import "./Results.css";

function Results({ className, ...props }) {
  return (
    <QuizListContext.Consumer>
      {(quizList) => (
        <QuizContext.Consumer>
          {(quiz) => <ResultsDisplay quiz={quiz} quizList={quizList} />}
        </QuizContext.Consumer>
      )}
    </QuizListContext.Consumer>
  );
}

export default Results;
