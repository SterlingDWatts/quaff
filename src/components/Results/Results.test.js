import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Results from "./Results";

describe("Results Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Results />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Results />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
