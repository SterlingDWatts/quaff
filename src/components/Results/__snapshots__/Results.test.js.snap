// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Component renders the UI as expected 1`] = `
<div
  className="ResultsDisplay"
/>
`;
